Sources/MessagingPush/MessagingPushImplementation.swift

```diff
@@ -113,6 +133,31 @@ internal class MessagingPushImplementation: MessagingPushInstance {
                                     data: MetricRequest(deliveryId: deliveryID, event: event, deviceToken: deviceToken,
                                                         timestamp: Date()))
     }
+    func getDefaultDeviceAttributes(completionHandler: @escaping([String: Any]?) -> Void) {
+        if !sdkConfigStore.config.autoTrackDeviceAttributes {
+            completionHandler(nil)
+            return
+        }
+        #if canImport(UIKit) && canImport(UserNotifications)
+        let deviceDetail = DeviceInfo()
+        let deviceOS = deviceDetail.osInfo
+        let deviceModel = deviceDetail.deviceInfo
+        let appVersion = deviceDetail.customerAppVersion
+        let sdkVersion = deviceDetail.sdkVersion
+        let deviceLocale = deviceDetail.deviceLocale.replacingOccurrences(of: "_", with: "-")
+        deviceDetail.pushSubscribed { isSubscribed in
+            let deviceAttributes = ["device_os": deviceOS,
+                                    "device_model": deviceModel,
+                                    "app_version": appVersion,
+                                    "cio_sdk_version": sdkVersion,
+                                    "device_locale": deviceLocale,
+                                    "push_subscribed": String(isSubscribed)]
+            completionHandler(deviceAttributes)
+        }
+        #else
+            completionHandler(nil)
+        #endif
```

Comment:
`import UIKit` is always done at file scope. If the condition` #if canImport(UIKit)` is added to the property `osInfo` value then it would work as expected going by your suggestion but the Linux, MacOs test cases will fail in CI/CD. 

I already tried implementing your suggestion but it results in failed test cases. Any other suggestion is always welcome !